numba/core/event.py

```diff
@@ -423,3 +426,42 @@ def on_exit(*exc_details):
 
         start_event(kind, data=data)
         yield
+
+
+def _write_chrome_trace(rec):
+    """ If chrome tracing is enabled,
+        write event information (process/thread id, event name,
+        and duration time to json file.
+    """
+    import os
+    import threading
+
+    pid = os.getpid()
+    tid = threading.get_ident()
+    evs = []
+    for ts, rec in rec.buffer:
+        data = rec.data
+        cat = str(rec.kind)
+        ph = 'B' if rec.is_start else 'E'
+        pid = pid
+        tid = tid
+        ts = ts
+        name = data['name']
+        args = data
+        ev = dict(
+            cat=cat, pid=pid, tid=tid, ts=ts, ph=ph, name=name, args=args,
+        )
+        evs.append(ev)
+    return evs
+
+
+if config.CHROME_TRACE:
```

Comment:
Should this be at module level? I wonder if Numba needs a process level hook mechanism.

Reply:
I don't think this will work with multiprocessing either, have been testing with:
```python
import os
import multiprocessing as mp
from numba import njit
import numpy as np

@njit
def go_fast(a):
    trace = 0.0
    for i in range(a.shape[0]):
        trace += np.tanh(a[i, i])
    return a + trace

def foo(q, arr):
    q.put(go_fast(arr))

if __name__ == '__main__':
    arr = np.arange(100).reshape(10, 10)
    mp.set_start_method('fork')
    q = mp.Queue()
    procs = []
    n = os.environ.get('NPROCS')
    assert n
    n = int(n)
    for i in range(n):
        procs.append(mp.Process(target=foo, args=(q, arr)))

    for proc in procs:
        proc.start()

    for proc in procs:
        proc.join()

    acc = 0
    for i in range(n):
        acc += q.get()

    assert np.allclose(acc[0, 0], 9 * n)
```
then:
```
NPROCS=4 NUMBA_CHROME_TRACE='/tmp/foo.json' python test_7537.py 
```
This may not matter/just be a limitation of this feature, in which case it should be added to the docs as such?